### PR TITLE
Update Changelog and test-plan for multiple download of omnisharp

### PR DESCRIPTION
**Pull request by akshita31**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,13 @@
 * Adds launch.json option to suppress Just-In-Time compiler optimizations.
 * Due to the previous two items and work from the .NET Team, it is now possible to easily debug into ASP.NET itself in projects running against .NET Core 2.1 preview 1. Support for debugging into all the managed code in .NET Core will come in future .NET Core 2.1 builds. Instructions are in the [wiki](https://github.com/OmniSharp/omnisharp-vscode/wiki/Debugging-into-the-.NET-Framework-itself).
 
+#### Alternative Omnisharp Support
+
+* Adds support to use the "omnisharp.path" option to download alternative copies of OmniSharp. The possible values for this option are:
```
Comment (rchande) [inline CHANGELOG.md]: Maybe describe the feature a little more?

"Enables the use of pre-release builds of Omnisharp. Downloading a pre-release build of Omnisharp allows the C# extension for VS Code to use features that have been merged into the "master" branch of omnisharp-roslyn [link] but that have not bee officially released"

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,13 @@
 * Adds launch.json option to suppress Just-In-Time compiler optimizations.
 * Due to the previous two items and work from the .NET Team, it is now possible to easily debug into ASP.NET itself in projects running against .NET Core 2.1 preview 1. Support for debugging into all the managed code in .NET Core will come in future .NET Core 2.1 builds. Instructions are in the [wiki](https://github.com/OmniSharp/omnisharp-vscode/wiki/Debugging-into-the-.NET-Framework-itself).
 
+#### Alternative Omnisharp Support
```
Comment (rchande) [inline CHANGELOG.md]: Not sure 'alternative' is the right word. Maybe "Nightly" or 'pre-release"?

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,13 @@
...
+
+* Adds support to use the "omnisharp.path" option to download alternative copies of OmniSharp. The possible values for this option are:
+    * "latest" - Use the latest CI build
+    * Version to use - The particular OmniSharp version the user desires like "1.29.2-beta.60"
```
Comment (rchande) [inline CHANGELOG.md]: "To use a specific pre-release build ..."
Describe where users can find these numbers, eg the folder omnisharp gets downloaded to

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,6 +11,13 @@
...
+    * Some absolute path - The absolute path on disk that points to the executables (like Omnisharp.exe in Windows) or binaries to be used to start the OmniSharp server
```
Comment (rchande) [inline CHANGELOG.md]: Copy whatever the option doc says for the path?

File: test-plan.md [hunk 5]
```diff
@@ -92,6 +92,19 @@ The easist way to verify that a project was successfully loaded is to open a .cs
...
 
+#### Omnisharp Options
+
+  #### omnisharp.useMono (for Linux)
+  For using this option, mono version greater than or equal to 5.2.0 must be installed. If that is not so, setting this option should give an error.
+  * If the option is not set, the OmniSharp path displayed in the "OmniSharp Log" should end with "run"
+  * If the option is set, the OmniSharp path as mentioned above should end with "OmniSharp.exe"
+
+  #### omnisharp.path
+  * Setting this to the default value(null) or empty string should start the OmniSharp server, and display the .omnisharp folder followed by the platform specific binary/executable as the OmniSharp path
```
Comment (rchande) [inline test-plan.md]: Display it where?

File: test-plan.md [hunk 6]
```diff
@@ -92,6 +92,19 @@ The easist way to verify that a project was successfully loaded is to open a .cs
...
+  * Setting this option to the (absolute)path that contains the platform specific binary/executable for Omnisharp, should start the OmniSharp server and display the same in the OmniSharp path
+  * Setting this to a particular version like "1.29.2-beta.60" should download and install the specified version packages and start the OmniSharp server and display the OmniSharp path that includes '.omnisharp/experimental/1.29.2-beta.60' followed by the platform specific binary/executable
+  * All the three possible values should start the OmniSharp server and display the appropriate OmniSharp path, with and without setting the useMono option on Linux
```
Comment (rchande) [inline test-plan.md]: Is there some behavior we should be testing around changing the version and doing the "restart omnisharp" command?

File: test-plan.md [hunk 7]
```diff
@@ -92,6 +92,19 @@ The easist way to verify that a project was successfully loaded is to open a .cs
 * In a project that uses globbing (.NET Core), use the VS Code file explorer to add a new file next to the csproj. Intellisense/sighelp/etc should be available in the new file
 * Add a new file and reference a type in it from a different file. Deleting from disk the file containing the referenced type  should produce error messages
 
+#### Omnisharp Options
+
+  #### omnisharp.useMono (for Linux)
```
Comment (TheRealPiotrP) [inline test-plan.md]: #### omnisharp.useMono (forLinux)
Determines which Mono installation is used to run Omnisharp. Defaults to false.
If true, the host's globally installed Mono is used. Mono must be version 5.2.0 or greater.
If false, the Mono bundled with Omnisharp is used.

Comment (akshita31) [reply test-plan.md]: @TheRealPiotrP As per this description, how will the people who test determine which version of mono is being used- the bundled one or the globally installed one. That is why I mentioned about using the "run" and the "Omnisharp.exe". Thoughts ?

File: test-plan.md [hunk 8]
```diff
@@ -92,6 +92,19 @@ The easist way to verify that a project was successfully loaded is to open a .cs
...
+  For using this option, mono version greater than or equal to 5.2.0 must be installed. If that is not so, setting this option should give an error.
+  * If the option is not set, the OmniSharp path displayed in the "OmniSharp Log" should end with "run"
+  * If the option is set, the OmniSharp path as mentioned above should end with "OmniSharp.exe"
+
+  #### omnisharp.path
```
Comment (TheRealPiotrP) [inline test-plan.md]: #### omnisharp.path
Determines which version of OmniSharp is used by the extension. Defaults to undefined.
* ##undefined## the extension uses the OmniSharp version it shipped with.
* ##Absolute Path## the extension uses a local copy of OmniSharp. The value must point to a directory which contains OmniSharp, typically a user's build output directory for the OmniSharp-Roslyn project. Example value: `C:\\omnisharp-roslyn\\artifacts\\publish\\OmniSharp.Stdio\\win7-x64\\OmniSharp.exe`.
* ##latest## the extension will download the latest build of OmniSharp produced by the OmniSharp-Roslyn repo.
* ##<version>## the extension will download a specific build of OmniSharp produced by the OmniSharp-Roslyn repo. Example value: `1.29.2-beta.60`.

File: CHANGELOG.md (hunk 4 above)
Comment (rchande) [reply CHANGELOG.md]: Disregard in favor of Peter's comment.

File: test-plan.md (hunk 5 above)
Comment (rchande) [reply test-plan.md]: Disregard in favor of Peter's comment.

File: test-plan.md (hunk 7 above)
Comment (TheRealPiotrP) [reply test-plan.md]: sorry, will fix up now :)

File: CHANGELOG.md (hunk 2 above)
Comment (TheRealPiotrP) [inline CHANGELOG.md]: #### Specify OmniSharp Version

File: CHANGELOG.md (hunk 3 above)
Comment (TheRealPiotrP) [reply CHANGELOG.md]: `latest` looks good. for the last one:
```
* "<version>" - Use a specific version of OmniSharp. Example: `1.29.2-beta.60`
```

File: CHANGELOG.md (hunk 4 above)
Comment (TheRealPiotrP) [reply CHANGELOG.md]: Some absolute path - the extension uses a local copy of OmniSharp. The value must point to a directory which contains OmniSharp, typically a user's build output directory for the OmniSharp-Roslyn project. Example: C:\\omnisharp-roslyn\\artifacts\\publish\\OmniSharp.Stdio\\win7-x64\\OmniSharp.exe.

File: test-plan.md (hunk 5 above)
Comment (TheRealPiotrP) [reply test-plan.md]: in the C# or O# channel?

File: test-plan.md (hunk 6 above)
Comment (TheRealPiotrP) [reply test-plan.md]: yes! we should also add a test for `latest`

File: test-plan.md [hunk 9]
```diff
@@ -92,6 +92,22 @@ The easist way to verify that a project was successfully loaded is to open a .cs
...
+  Setting this path to any of the values as listed below, should start the OmniSharp server and display the correct OmniSharp path in the `OmniSharp Log`(View --> Output--> OmniSharp Log).
+  * undefined - OmniSharp server must start using the copy of omnisharp shipped with the extension, that is, the OmniSharp path must be the extension path, followed by .omnisharp followed by the platform-specific executable.
+  * Some absolute path - OmniSharp server must start using the local copy of OmniSharp pointed to by the path and the same must be displayed as the Omnisharp path. Example:C:\omnisharp-roslyn\artifacts\publish\OmniSharp.Stdio\win7-x64\OmniSharp.exe. 
+  * "<version>" - The specified version packages must be downloaded and installed (The status of download/install can be seen in the `C#` log). If the installation is successful, the server must start and the OmniSharp path must include the .omnisharp/experimental folder followed by the version name and the executable. Eg: If the version is `1.29.2-beta.60`, the path displayed on Windows should be `.omnisharp/experimental/1.29.2-beta.60/OmniSharp.exe`.
```
Comment (rchande) [inline test-plan.md]: Just "version"--no "packages"

**Pull request merged**